src/Workleap.DomainEventPropagation.Abstractions/DomainEventAttribute.cs

```diff
@@ -3,20 +3,32 @@
 [AttributeUsage(AttributeTargets.Class, Inherited = false)]
 public sealed class DomainEventAttribute : Attribute
 {
-    public DomainEventAttribute(string name)
+    public DomainEventAttribute(string name) : this(name, EventSchema.EventGridEvent)
+    {
+    }
+
+    public DomainEventAttribute(string name, EventSchema eventSchema = EventSchema.EventGridEvent)
     {
         if (name == null)
         {
             throw new ArgumentNullException(nameof(name));
         }
 
+        if (eventSchema == null)
+        {
+            throw new ArgumentNullException(nameof(eventSchema));
+        }
```

Comment:
I would remove the validation altogether given `eventSchema` will never have a nullable value.

```suggestion
```